app/workers/functions.php

```diff
@@ -277,7 +275,7 @@ function: $function,
                         trigger: 'event',
                         event: $events[0],
                         eventData: \is_string($eventData) ? $eventData : \json_encode($eventData),
-                        user: $user,
+                        user: $user->isEmpty() ? null : $user,
```

Comment:
In line 240, you can define `$user` as follows, and avoid the comparisons you do in both cases:

`$user = $payload['user']`